metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -451,6 +451,110 @@ class MetalsLanguageServer(
       definitionResult(position).locations
     }
 
+  @JsonRequest("textDocument/typeDefinition")
+  def typeDefinition(
+      position: TextDocumentPositionParams
+  ): CompletableFuture[util.List[Location]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Goto type definition is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/implementation")
+  def implementation(
+      position: TextDocumentPositionParams
+  ): CompletableFuture[util.List[Location]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Goto implementation is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/hover")
+  def hover(params: TextDocumentPositionParams): CompletableFuture[Hover] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Hover is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/documentHighlight")
+  def documentHighlights(
+      params: TextDocumentPositionParams
+  ): CompletableFuture[Hover] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Document highlights is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/documentSymbol")
+  def documentSymbol(
+      params: DocumentSymbolParams
+  ): CompletableFuture[util.List[DocumentSymbol]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Document symbols is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/formatting")
+  def documentSymbol(
+      params: DocumentFormattingParams
+  ): CompletableFuture[util.List[TextEdit]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Code formatting is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/rename")
+  def documentSymbol(
+      params: RenameParams
+  ): CompletableFuture[WorkspaceEdit] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Rename symbol is not supported.")
+      null
+    }
+
+  @JsonRequest("textDocument/references")
+  def references(
+      position: ReferenceParams
+  ): CompletableFuture[util.List[Location]] =
+    CompletableFutures.computeAsync { _ =>
+      scribe.warn("Find references is not supported.")
```

Comment:
Super minor but I'd use the request identifier in the warning message "textDocument/references is not implemented"

Reply:
I support this. "textDocument/references is not implemented **yet**" 😄

Reply:
I agree with including the method name, I'm on the fence to promise what will get implemented.

Reply:
I'm just joking 😉